src/core/client/embed/StreamEmbed.ts

```diff
@@ -43,139 +40,176 @@ export interface StreamEmbedConfig {
 }
 
 export class StreamEmbed {
-  private config: StreamEmbedConfig;
-  private pymControl?: PymControl;
-  private pymControlFactory: PymControlFactory;
+  /**
+   * Every interval rounded to this value in ms will be passed when creating the
+   * stream embed iframe to ensure that it is loaded fresh.
+   */
+  private readonly requestExpiryInterval = 15 * 60 * 1000; // 15 minutes
+
+  /**
+   * control provides an interface to the iFrame controller.
+   */
+  private readonly control: FrameControl;
+
+  /**
+   * eventEmitter provides an interface to events emitted by Coral.
+   */
+  private readonly eventEmitter: EventEmitter2;
+
   private ready = false;
-  private cancelAutoRender: OnIntersectCancellation | null = null;
+  private clearAutoRender: OnIntersectCancellation | null = null;
 
   constructor(
     config: StreamEmbedConfig,
-    pymControlFactory = defaultPymControlFactory
+    controlFactory: FrameControlFactory = defaultPymControlFactory
   ) {
-    this.config = config;
-    this.pymControlFactory = pymControlFactory;
+    const element = document.getElementById(config.id);
+    if (!element) {
+      throw new Error(`element ${config.id} was not found`);
+    }
+
+    // Save a reference to the event emitter used by the application.
+    this.eventEmitter = config.eventEmitter;
+
+    // When the config emits that we're ready, then mark us as ready.
+    config.eventEmitter.once("ready", () => {
+      this.ready = true;
+    });
+
+    // Create the decorators that will be used by the controller.
+    const decorators: ReadonlyArray<Decorator> = [
+      withIOSSafariWidthWorkaround,
+      withAutoHeight,
+      withClickEvent,
+      withSetCommentID,
+      withEventEmitter(config.eventEmitter, config.enableDeprecatedEvents),
+      withLiveCommentCount(config.eventEmitter),
+      withPymStorage(localStorage, "localStorage"),
+      withPymStorage(sessionStorage, "sessionStorage"),
+      withConfig({
+        accessToken: config.accessToken,
+        bodyClassName: config.bodyClassName,
+      }),
+      withKeypressEvent,
+      withRefreshAccessToken(config.refreshAccessToken),
+    ];
+
+    const query = stringifyQuery({
+      storyID: config.storyID,
+      storyURL: config.storyURL,
+      commentID: config.commentID,
+      customCSSURL: config.customCSSURL,
+
+      // Add the version to the query string to ensure that every new version of
+      // the stream will cause stream pages to cache bust.
+      v: process.env.TALK_VERSION ? process.env.TALK_VERSION : "dev",
```

Comment:
Is it okay to expose our version from a security pov?

Reply:
The version is already exposed in the application releases as it is, this isn't revealing anything new.